tools/compile-perf/native/api-driver.cpp

```
@@ -97,6 +102,64 @@ struct Timers
+    return resident * (sysconf(_SC_PAGESIZE) / 1024);
+#endif
+}
+
+// Memory deltas recorded around selected phases, reported next to the timers
+// as "[MEM] name\tNNNkb" lines that bench.py parses into the record's
+// `memory` dict. The RSS growth across createGlobalSession is the metric of
+// shader-slang/slang#9817. Reuses Timers as the accumulator (totalMs holds
+// KB); reportMemDeltas() runs wherever timers.report() does.
+static Timers g_memDeltas;
+
+static void recordSessionCreateRss(long rssBefore)
```

Cppcheck (2.21.0)

[style] The function 'attach' is never used.

(unusedFunction)